gems/smithy-client/lib/smithy-client/plugins/user_agent.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Plugins
+      class UserAgent < Plugin
+        class Handler < Client::Handler
+          def call(context)
+            set_user_agent(context)
+            @handler.call(context)
+          end
+
+          def set_user_agent(context)
+            context.http_request.headers['User-Agent'] = UserAgent.new(context).to_s
+          end
+
+          class UserAgent
+            def initialize(context)
+              @context = context
+            end
+
+            def to_s
+              ua = "smithy-ruby/#{File.read(File.expand_path('../../../VERSION', __dir__)).strip}"
```

Comment:
There should be a constant for this .. Smithy::Client::VERSION if it isn't already there.